From ac481b2e85c9daef50a5e4a652f7c295eb4baf25 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sat, 18 Jan 2020 21:56:16 +0200
Subject: [PATCH] [SW-8228] arcboot: fuse_read() should use udelay()

Current loop is just not correct, single nop is not taking uSec

Change-Id: Idac8d59a90269775520024ade3dc1c6802434639
---

diff --git a/common/efuse.c b/common/efuse.c
index 35acc78..429f466 100644
--- a/common/efuse.c
+++ b/common/efuse.c
@@ -7,6 +7,7 @@
 
 #include <unistd.h>
 #include "arcboot.h"
+#include "delay.h"
 
 #define FUSE_TIMEOUT_US		100000
 #define FUSE_DELAY_US		20
@@ -20,7 +21,7 @@
  */
 int fuse_read(u32 word, u32 *val)
 {
-	int i, offset, delay;
+	int i, offset;
 
 	if (word > 64) {
 		return -1;
@@ -30,7 +31,7 @@
 
 	for (i=0; i < FUSE_RETRIES; i++)
 		if (regmap_readl(mmPSOC_EFUSE_READ))
-			for(delay=FUSE_DELAY_US; delay > 0; delay--) asm("nop");
+			udelay(FUSE_DELAY_US);
 		else
 			break;
 
